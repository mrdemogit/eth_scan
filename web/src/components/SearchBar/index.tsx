import React from 'react';
import styled from 'styled-components/macro';

const SearchInputStyled = styled.input`
  padding: 1rem 0;
  background: #353535;
  border: none;
  box-shadow: inset 0 0 0.5px #f7f7f7;
  width: 100%;
  font-size: 1rem;
  text-align: center;
  color: #fff;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

interface Props {
  onChange: (e: string) => void;
  onBlur: (e: string) => void;
  placeholder: string;
}

const SearchBar: React.FC<Props> = ({ onChange, onBlur, ...props }) => (
  <SearchInputStyled
    {...props}
    onChange={e => onChange(e.target.value)}
    onBlur={e => onBlur(e.target.value)}
  />
);

export default SearchBar;
