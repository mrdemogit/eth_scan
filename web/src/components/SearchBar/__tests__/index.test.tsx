import React from 'react';
import { render } from 'react-testing-library';
import SearchBar from '../index';

describe('SearchBar', () => {
  it('Render default SearchBar', () => {
    const { container } = render(
      <SearchBar
        placeholder="placeholder"
        onChange={() => {}}
        onBlur={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
