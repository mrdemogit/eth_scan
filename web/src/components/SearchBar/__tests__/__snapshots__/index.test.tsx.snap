// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Render default SearchBar 1`] = `
<div>
  <input
    class="SearchBar__SearchInputStyled-sc-1aqayj1-0 euIjYj"
    placeholder="placeholder"
  />
</div>
`;
